### PR TITLE
Family test badges (EXPOSUREAPP-12315)

### PR DESCRIPTION
**Pull request by mtwalli**
## Changes:
- Badges logic for family tests
- Registration flow and requesting Dcc is working
- Fix navigation from family tile / fix ordering of the tile in Home Screen

## Testing
- You can register a family test , and it is saved properly, badges are shown in the Home Screen / tile 
## Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12315

### CODE REVIEW

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/core/repository/CoronaTestProcessorTest.kt [hunk 1]
```diff
@@ -37,6 +37,14 @@ class CoronaTestProcessorTest : BaseTest() {
         testResult = CoronaTestResult.PCR_OR_RAT_PENDING
     )
 
+    private val updatedTest = CoronaTest(
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/familytest/core/repository/CoronaTestProcessorTest.kt:40]: this is unused and can be removed

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/CoronaTestProcessor.kt [hunk 2]
```diff
@@ -98,15 +98,34 @@ class CoronaTestProcessor @Inject constructor(
...
+        return when (qrCode) {
+            is CoronaTestQRCode.PCR -> {
+                val dateOfBirth = qrCode.dateOfBirth
+                val dateOfBirthKey = if (qrCode.isDccConsentGiven && dateOfBirth != null) {
+                    DateOfBirthKey(qrCode.qrCodeGUID, dateOfBirth)
+                } else null
+
+                RegistrationRequest(
+                    key = qrCode.qrCodeGUID,
+                    dateOfBirthKey = dateOfBirthKey,
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/CoronaTestProcessor.kt:110]: override val registrationIdentifier: String
            get() = qrCodeGUID

This should have worked for all types, or did it not?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/storage/FamilyTestDatabase.kt [hunk 3]
```diff
@@ -88,10 +84,10 @@ interface FamilyCoronaTestDao {
     suspend fun delete(entity: FamilyCoronaTestEntity)
 
     @Query("SELECT * FROM family_corona_test WHERE moved_to_recycle_bin_at_millis IS NULL")
-    fun getAllActive(): Flow<List<FamilyCoronaTestEntity>>
+    fun getAllActive(): Flow<List<FamilyCoronaTestEntity?>>
```
[the remark this answers is not in the document]
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/storage/FamilyTestDatabase.kt:87]: if parsing  fails ,see  `FamilyCoronaTestConverter`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/CoronaTestProcessor.kt (hunk 2 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/core/repository/CoronaTestProcessor.kt:110]: it did not

**Review by AlexanderAlferov — APPROVED** [review]

**Review by chiljamgossow — APPROVED** [review]

**Pull request merged**